scripts/gen_vpc_ip_limits.go

```diff
@@ -128,65 +134,76 @@ func describeRegions() []string {
 func describeInstanceTypes(region string, eniLimitMap map[string]vpc.InstanceTypeLimits) {
 	log.Infof("Describing instance types in region=%s", region)
 
-	// Get session
-	sess := session.Must(session.NewSessionWithOptions(session.Options{
-		SharedConfigState: session.SharedConfigEnable,
-		Config:            *aws.NewConfig().WithRegion(region),
-	}))
-	_, err := sess.Config.Credentials.Get()
+	// Create context
+	ctx := context.Background()
+
+	// Load AWS configuration with specific region
+	cfg, err := config.LoadDefaultConfig(ctx,
+		config.WithRegion(region),
+	)
 	if err != nil {
-		log.Fatalf("Failed to get session credentials: %v", err)
+		log.Fatalf("Failed to load configuration: %v", err)
 	}
-	svc := ec2.New(sess)
-	describeInstanceTypesInput := &ec2.DescribeInstanceTypesInput{}
 
-	for {
-		output, err := svc.DescribeInstanceTypes(describeInstanceTypesInput)
+	// Create EC2 client
+	client := ec2.NewFromConfig(cfg)
+
+	paginator := ec2.NewDescribeInstanceTypesPaginator(client, &ec2.DescribeInstanceTypesInput{})
+
+	// Iterate through all pages
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
 		if err != nil {
 			log.Fatalf("Failed to call EC2 DescribeInstanceTypes: %v", err)
 		}
+
 		// We just want the type name, ENI and IP limits
 		for _, info := range output.InstanceTypes {
 			// Ignore any missing values
-			instanceType := aws.StringValue(info.InstanceType)
+			instanceType := string(info.InstanceType)
+
 			// only one network card is supported, so use the MaximumNetworkInterfaces from the default card if more than one are present
 			var eniLimit int
 			if len(info.NetworkInfo.NetworkCards) > 1 {
-				eniLimit = int(aws.Int64Value(info.NetworkInfo.NetworkCards[*info.NetworkInfo.DefaultNetworkCardIndex].MaximumNetworkInterfaces))
+				eniLimit = int(*info.NetworkInfo.NetworkCards[*info.NetworkInfo.DefaultNetworkCardIndex].MaximumNetworkInterfaces)
```

Comment:
v2 has `aws.ToString` etc. which is the same thing as `aws.StringValue` from v1

Reply:
> instanceType := string(info.InstanceType)
This wasn't a conversion from *string to string. a new type InstanceType (type InstancetType string) is introduced, which is we treat it as string, so instead of changing the entire program I used it as string.